### DISCUSSION TITLE
Do we have "or" condition for elements collection

### DISCUSSION THREAD

**Discussion by VladyslavZagorodnyi** [selenide/selenide#2681] [Q&A]
Hello. I identified interesting case, let's assume I have elements collection and I need to check text of elements inside collection. It may look like:
```
ElementsCollection collectionOfButtons = $$("div");
collectionOfButtons.shouldHave(texts("Button A", "Button B"));
```
And it looks fine, short and easy, but let's assume that sometimes button B can be changed to button C (dummy example, just because of test logic or because some env limitation ... )

So how can I make or logic? It could be very useful to make something like this:
```
collectionOfButtons.shouldHave(
    texts("Button A", "Button B")
).or(
    texts("Button A", "Button C")
);
```
so if there will NOT be "Button A", "Button B"  but will BE "Button A", "Button C" - test will finish successfully. 

Maybe somebody faced such cases previously, let me know if there is a good solution for such requirements (without adding if else ...)

**Comment by rosolko** [comment]
or is also an if-else-development and it is not a good practice. better to create two atomic tests for both Button A + Button B and Button A + Button C cases

**Comment by VladyslavZagorodnyi** [comment]
@asolntsev maybe you could step into this conversation?

**Reply by asolntsev** [reply]
@VladyslavZagorodnyi That's true: currently we don't have method "or" for collections. 
A totally agree with @rosolko : it's a form of "if-else" logic which should be avoided in tests. 

P.S. Yes, we have "or" method for single element, and I never liked it for the same reason. :)
And yes, we could add similar "or" method for connections. But I still recommend to replace this method by two different methods.
